### PR TITLE
Util: add batchArray and batchIterator methods

### PR DESCRIPTION
**Pull request by blueridger**
# Description
These methods help batch Arrays and Iterators (split the contents up into chunks).

The `batchIterator` method is used in the follow-up PR: https://github.com/sourcecred/sourcecred/pull/3258

The `batchArray` just because it might be useful at some point soon as we run into more scalability concerns in places we don't have iterators in use yet.

# Test Plan
unit tests + https://github.com/sourcecred/sourcecred/pull/3258

### CODE REVIEW

File: packages/sourcecred/src/util/batch.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+Example:
+  batch([1,2,3,4,5], 2) = [[1,2], [3,4], [5]]
+*/
+export function batchArray<T>(
+  array: $ReadOnlyArray<T>,
+  batchSize: number
+): $ReadOnlyArray<$ReadOnlyArray<T>> {
+  const result = [];
+  let backlog = array;
+  while (backlog.length) {
```
Comment (amrro) [inline packages/sourcecred/src/util/batch.js:15]: nit: I've tried to come up with a functional way without worrying about length that satisfies most of corner cases, then found `reduce` function: it's not well-documented on flow.
```js
function batchArray<T>(array: $ReadOnlyArray<T>, batchSize: number): $ReadOnlyArray<$ReadOnlyArray<T>> {
  return array.reduce(
    (previousValue, currentItem, idx) =>
      (idx % batchSize === 0 // simply, should we start a new array item or not.
        ? previousValue.push([currentItem]) // yes? okay, wrap the current item into array.
        : previousValue[previousValue.length - 1].push(currentItem)) &&
      previousValue,
    <T>[] // initial value incorrect syntax
  );
}
``` 

consider it as an alternative, but _personally_ I think loops are source of evil.

File: packages/sourcecred/src/util/batch.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    result.push(backlog.slice(0, batchSize));
```
Comment (amrro) [inline packages/sourcecred/src/util/batch.js:16]: pedantically speaking: what if the `batchSize` is `0` or with minus value? 
I think, the `while` loop will run forever, if the `batchSize` is smaller than `1`, for sure if `0`.

File: packages/sourcecred/src/util/batch.test.js [hunk 3]
```diff
@@ -0,0 +1,80 @@
+// @flow
+
+import {batchArray, batchIterator} from "./batch";
+
+describe("src/util/batch", () => {
+  describe("batch", () => {
+    it("batches an array bigger than the batch size", () => {
```
Comment (amrro) [inline packages/sourcecred/src/util/batch.test.js:7]: Suggestion: add tests for corner cases of `batchSize`.

**Review by amrro — CHANGES_REQUESTED** [review]

File: packages/sourcecred/src/util/batch.js (hunk 1 above)
Comment (blueridger) [reply packages/sourcecred/src/util/batch.js:15]: Clever, I think that reduces readability significantly. It's more understandable as written--a batching method working at the granularity of batches.

File: packages/sourcecred/src/util/batch.test.js (hunk 3 above)
Comment (blueridger) [reply packages/sourcecred/src/util/batch.test.js:7]: accepted

File: packages/sourcecred/src/util/batch.js (hunk 2 above)
Comment (blueridger) [reply packages/sourcecred/src/util/batch.js:17]: Good point. I'll add a conditional throw.

**Review by amrro — APPROVED** [review]
